### PR TITLE
Adding GCP support

### PR DESCRIPTION
**Pull request by AleksCallebat**
Added GCP Support. 

The repo has 3 folder : 
- One to create an SA. This is a prerequisite for the other 2 repos to function
- One to deploy with a managed VPC
- One to deploy to a customer VPC

### CODE REVIEW

File: examples/gcp-sa-provisionning/init.tf
Comment (nkvuong) [inline examples/gcp-sa-provisionning/init.tf:3]: you don't need databricks provider to provision a GCP SA

File: examples/gcp-byovpc/vpc.tf
Comment (nkvuong) [inline examples/gcp-byovpc/vpc.tf:9]: these IP ranges shouldn't be hardcoded

File: examples/gcp-lakehouse/workspace.tf
Comment (nkvuong) [inline examples/gcp-lakehouse/workspace.tf:18]: delete commented out code

File: examples/gcp-byovpc/workspace.tf
Comment (nkvuong) [inline examples/gcp-byovpc/workspace.tf:1]: please run `terraform fmt -recursive` to format all the tf files

File: examples/gcp-lakehouse/readme.md
Comment (nkvuong) [inline examples/gcp-lakehouse/readme.md:1]: should the name of this folder be gcp-basic

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf
Comment (alexott) [inline examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf:1]: Please don't use paths with spaces

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf
Comment (alexott) [inline examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf:6]: it's not recommended to use version specification inside the module, or at lease provide ranges: https://developer.hashicorp.com/terraform/language/modules/develop/providers#provider-version-constraints-in-modules

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/readme.md
Comment (alexott) [inline examples/gcp-lakehouse/modules/gcp - sa provisionning/readme.md:1]: please make naming uniform - other modules has `README.md` as a name

**Review by alexott — CHANGES_REQUESTED** [review]

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf
Comment (AleksCallebat) [reply examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf:1]: corrected - thanks

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf
Comment (AleksCallebat) [reply examples/gcp-lakehouse/modules/gcp - sa provisionning/init.tf]: OK - removed version altogether as I don't think it would be highly impactful

File: examples/gcp-lakehouse/modules/gcp - sa provisionning/readme.md
Comment (AleksCallebat) [reply examples/gcp-lakehouse/modules/gcp - sa provisionning/readme.md:1]: Done - thanks for the review

File: examples/gcp-lakehouse/modules/gcp-basic/variables.tf
Comment (alexott) [inline examples/gcp-lakehouse/modules/gcp-basic/variables.tf:1]: Do we have any outputs to return back? Like, workspace URL?

File: examples/gcp-lakehouse/modules/gcp-sa-provisionning/main.tf
Comment (alexott) [inline examples/gcp-lakehouse/modules/gcp-sa-provisionning/main.tf:6]: it makes sense to put all outputs into a separate file, like [here](https://github.com/databricks/terraform-databricks-examples/blob/main/modules/aws-workspace-basic/outputs.tf), so registry will show them correctly: https://registry.terraform.io/modules/databricks/examples/databricks/latest/submodules/aws-workspace-basic

File: examples/gcp-lakehouse/modules/gcp-basic/variables.tf
Comment (AleksCallebat) [reply examples/gcp-lakehouse/modules/gcp-basic/variables.tf:1]: yes - workspace URL, and token

File: examples/gcp-lakehouse/modules/gcp-sa-provisionning/main.tf
Comment (AleksCallebat) [reply examples/gcp-lakehouse/modules/gcp-sa-provisionning/main.tf:6]: makes sense. done. Should I break it down apart more to match the granularity of the aws workspace one ?

**Review by nfx — APPROVED** [review]

**Comment by yessawab** [comment]
@alexott @nkvuong can you approve this PR if all is good ? 
I will restructure it later to build modules out of this example.

File: examples/gcp-lakehouse/modules/gcp-sa-provisionning/README.md
Comment (nkvuong) [inline examples/gcp-lakehouse/modules/gcp-sa-provisionning/README.md:1]:
```suggestion
# Provisioning a Google Service Account that can be used to deploy Databricks workspace on GCP
```

File: examples/gcp-lakehouse/modules/gcp-sa-provisionning/README.md
Comment (nkvuong) [inline examples/gcp-lakehouse/modules/gcp-sa-provisionning/README.md:4]:
```suggestion
In this template, we show how to deploy a service account that can be used to deploy Databricks workspace on GCP.
```

File: examples/gcp-lakehouse/modules/gcp-basic/README.md
Comment (nkvuong) [inline examples/gcp-lakehouse/modules/gcp-basic/README.md:4]:
```suggestion
In this template, we show how to deploy a workspace with managed VPC.
```

**Review by nkvuong — CHANGES_REQUESTED** [review]

**Review by nkvuong — APPROVED** [review]

**Comment by nkvuong** [comment]
@yessawab need to resolve conflicts before we can merge this PR

**Review by alexott — APPROVED** [review]

**Comment by alexott** [comment]
@AleksCallebat can you please rebase onto the latest `main` ?

**Comment by AleksCallebat** [comment]
Rebase done - no conflicts anymore, seems ready to merge to me

File: examples/adb-vnet-injection/variables.tf
Comment (alexott) [inline examples/adb-vnet-injection/variables.tf:18]: why this? it's better to leave it small

**Review by alexott — CHANGES_REQUESTED** [review]
please address a comment about removed links in README

File: examples/adb-vnet-injection/variables.tf
Comment (AleksCallebat) [reply examples/adb-vnet-injection/variables.tf]: I am not sure what caused this specific merge conflict as this isn't a part I touched. I agree with you on this.

**Review by alexott — APPROVED** [review]

**Pull request merged**
